base/show.jl

```diff
@@ -1135,6 +1135,7 @@ end
 
 function show_unquoted_quote_expr(io::IO, @nospecialize(value), indent::Int, prec::Int, quote_level::Int)
     if isa(value, Symbol) && !(value in quoted_syms)
+        value = value::Symbol
```

Comment:
I found this change and the one below it surprising. Seems like the compiler should figure this one out.

Reply:
I think it only handles the case where the `isa` is the last conditional so writing it like

```
if isa(value, Symbol)
    if !(value in quoted_syms)
        ...
    end
end
```

might help it.